stdlib/private/SwiftReflectionTest/SwiftReflectionTest.swift

```diff
@@ -154,9 +154,13 @@ extension Section {
 internal func getReflectionInfoForImage(atIndex i: UInt32) -> ReflectionInfo? {
 #if INTERNAL_CHECKS_ENABLED
   return _getMetadataSection(UInt(i)).map { rawPointer in
-    let name = _getMetadataSectionName(rawPointer)
+    let cName = _copyMetadataSectionName(rawPointer)
+    defer {
+      cName.deallocate()
```

Comment:
Is there a guarantee that `deallocate` will use the correct memory function for the matching `strdup`?  Note that there are DLL specific heaps potentially at play.

Reply:
Mike commented on the same thing—technically, as currently implemented, this does work, but it's kind of evil.

Reply:
GMTA!